src/uproot/writing/interpret.py

```diff
@@ -0,0 +1,306 @@
+        y_errors(None or 1D numpy.ndarray): Symethrical values of errors for corresponding y value (length of y_errors has to be the same as x and y)
+        x_errors_low(None or 1D numpy.ndarray): Asymmetrical lower values of errors for corresponding x value (length of x_errors_low has to be the same as x and y)
+        x_errors_high(None or 1D numpy.ndarray): Asymmetrical upper values of errors for corresponding x value (length of x_errors_high has to be the same as x and y)
+        y_errors_low(None or 1D numpy.ndarray): Asymmetrical lower values of errors for corresponding y value (length of y_errors_low has to be the same as x and y)
+        y_errors_high(None or 1D numpy.ndarray): Asymmetrical upper values of errors for corresponding y value (length of y_errors_high has to be the same as x and y)
+        title (str): Title of the histogram.
+        xAxisLabel (str): Label of the X axis.
+        yAxisLabel (str): Label of the Y axis.
+        minY (None or float): Minimum value on the Y axis to be shown, if set to None then minY=min(y)
+        maxY (None or float): Maximum value on the Y axis to be shown, if set to None then maxY=max(y)
+        lineColor (int): Line color. (https://root.cern.ch/doc/master/classTAttLine.html)
+        lineStyle (int): Line style.
+        lineWidth (int): Line width.
+        fillColor (int): Fill area color. (https://root.cern.ch/doc/master/classTAttFill.html)
+        fillStyle (int): Fill area style.
+        markerColor (int): Marker color. (https://root.cern.ch/doc/master/classTAttMarker.html)
+        markerStyle (int): Marker style.
+        markerSize (float): Marker size.
+
+    WARNING! This function only works for TGraph, because serialization of TGraphErrors and TGraphAsymmErrors is not implemented yet.
+
+    Function that converts arguments into TGraph, TGraphErrors or TGraphAsymmErros based on the given arguments.
+    When all errors are unspecified, detected object is TGraph.
+    When x_errors, y_errors are specified, detected object is TGraphErrors.
+    When x_errors_low, x_errors_high, y_errors_low, y_errors_high are specified, detected object is TGraphAsymmErrors.
+    Note that both x_errors, y_errors need to be specified or set to None.
+    The same rule applies to x_errors_low, x_errors_high, y_errors_low, y_errors_high.
+    Also can't specify x_errors, y_errors and x_errors_low, x_errors_high, y_errors_low, y_errors_high at the same time.
+    All rules are designed to remove any ambiguity.
+    """
+
+    sym_errors = [x_errors, y_errors]
+    sym_errors_bool = [err is not None for err in sym_errors]
+
+    asym_errors = [x_errors_low, x_errors_high, y_errors_low, y_errors_high]
+    asym_errors_bool = [err is not None for err in asym_errors]
+
+    tgraph_type = "TGraph"
+
+    # Detecting which type of TGraph to chose
+    if any(sym_errors_bool):
+        if not all(sym_errors_bool):
+            raise ValueError("Have to specify both x_errors and y_errors")
```

Comment:
Error messages are lower case with no end-punctuation (period), but I'm also rewriting it to tell the user which high-level function has failed. If this is called by a library that the user didn't write, they might not have called `uproot.to_TGraph` themselves, and they might not even know that it exists. Mentioning it here helps them to debug.

```suggestion
            raise ValueError("uproot.to_TGraph requires both x_errors and y_errors")
```